torch/csrc/autograd/functions/accumulate_grad.cpp

```diff
@@ -59,5 +60,34 @@ auto AccumulateGrad::apply(variable_list&& grads) -> variable_list {
   return variable_list();
 }
 
+#ifdef COMPILED_AUTOGRAD
+void AccumulateGrad::compiled_args(CompiledNodeArgs& args) {
+  at::Tensor& grad = variable.mutable_grad();
+  args.collect(grad.defined());
+  if (grad.defined()) {
+    args.collect(grad);
+  }
+  args.set_grad_target(variable);
+}
+
+variable_list AccumulateGrad::apply_with_saved(
+    const variable_list& inputs,
+    SwapSavedVariables& saved) {
+  at::Tensor& grad = variable.mutable_grad();
+
+  // TODO(jansel): This is the wrong place for this, as I don't see this in the
+  // above apply. Where does this implicit sum happen in eager mode?
```

Comment:
It's there, via `clone_obey_contract` and `obeys_layout_contract`. If the gradient needed to be expanded, it would not obey layout contract and clone_obey_contract would make sure this does the right thing. You probably should figure out how to use those functions to implement the version here, since you won't get correct striding as written here.